pyls/workspace.py

```diff
@@ -16,11 +20,59 @@
 RE_END_WORD = re.compile('^[A-Za-z_0-9]*')
 
 
+def get_submodules(mod):
+    """Get all submodules of a given module"""
+    def catch_exceptions(module):
+        pass
+    try:
+        m = __import__(mod)
+        submodules = [mod]
+        submods = pkgutil.walk_packages(m.__path__, m.__name__ + '.',
+                                        catch_exceptions)
+        for sm in submods:
+            sm_name = sm[1]
+            submodules.append(sm_name)
+    except ImportError:
+        return []
+    except Exception:
+        return [mod]
+    return submodules
+
+
+def get_preferred_submodules():
+    mods = ['numpy', 'scipy', 'sympy', 'pandas',
```

Comment:
@gatesn Do you agree to preload all these modules?

Reply:
Supposing we do preload all of these modules. What's the Jedi performance like after the initial import? I've never run into an issue with Jedi after the first import.

Reply:
Well, the preloading only affects Rope. With respect to jedi, I found that it is possible also to preload modules, I'll make some comparisons

Reply:
Maybe @ccordoba12 has more information about which features has Rope that Jedi doesn't

Reply:
@gatesn I've found that rope is only faster on the preloaded modules than Jedi

Reply:
Yep, Rope is *significantly* faster to complete compiled modules than Jedi.

@andfoy, could we expose this value somehow, in case we want to add more modules to it in Spyder?

Reply:
@ccordoba12, unfortunately, under the LSP specs, we cannot do such thing

Reply:
@andfoy can't we take them as configuration options? https://github.com/palantir/python-language-server/blob/develop/vscode-client/package.json#L22

Reply:
@gatesn Yes we can, but those configuration settings are only used if LSP is launched on VSCode, isn't it?

Reply:
I think most clients have some interpretation of these configuration options.